### DISCUSSION TITLE
Support the management of dependencies in compressed packages

### DISCUSSION THREAD

**Discussion by jadendxc** [npm/rfcs#450] [RRFC]
Can npm package the .js file like a Java jar package, which can greatly improve the efficiency of dependent transmission

**Comment by ljharb** [comment]
It packages it in a tarball, is that not sufficient?

**Comment by thescientist13** [comment]
I think one important distinction here is that you can run a .jar file because that is how Java was designed, right?  Currently JS (browser or server) has no standard mechanism to run from an archive file format like Java can, though there is discussion on a proposal for that - https://github.com/WICG/webpackage

Otherwise as @ljharb said, a tar / zip is the next best thing (IMO)
